tests/src/test/scala/com/nvidia/spark/rapids/PrioritySemaphoreSuite.scala

```diff
@@ -88,4 +88,28 @@ class PrioritySemaphoreSuite extends AnyFunSuite {
     latch.await(1, TimeUnit.SECONDS)
     assert(results.asScala.toList == List(3, 2, 1))
   }
+
+  test("low priority thread cannot surpass high priority thread") {
+    val semaphore = new TestPrioritySemaphore(10)
+    val latch = new CountDownLatch(1)
+    semaphore.acquire(5, 0)
+    val t = new Thread(() => {
+      try {
+        semaphore.acquire(10, 2)
+      } finally {
+        semaphore.release(10)
+        latch.countDown()
+      }
+    })
+    t.start()
+    Thread.sleep(100)
+
+    // Here, there should be 5 available permits, but a thread with higher priority (2)
+    // is waiting to acquire, therefore we should get rejected here
+    assert(!semaphore.tryAcquire(5, 0))
+    semaphore.release(5)
+    latch.await(1, TimeUnit.SECONDS)
```

Comment:
Curious why the use of a countdown latch rather than joining on the thread, since the latch is at the end of the thread's execution.

Reply:
I don't have a good reason actually, let me try joining on thread